openedx/core/djangoapps/credit/tests/test_models.py

```diff
@@ -83,3 +90,96 @@ def add_credit_course(self):
         credit_course = CreditCourse(course_key=self.course_key, enabled=True)
         credit_course.save()
         return credit_course
+
+
+@ddt.ddt
+class CreditRequestModelTests(TestCase):
+    """
+    Tests for models used to track credit requests and status.
+    """
+
+    USERNAME = u"rön"
+    PROVIDER_ID = u"høgwarts"
+
+    def setUp(self):
+        """Create a course and provider. """
+        super(CreditRequestModelTests, self).setUp()
+        self.course_key = CourseKey.from_string("edX/DemoX/Demo_Course")
+        self.credit_course = CreditCourse.objects.create(course_key=self.course_key)
+        self.credit_provider = CreditProvider.objects.create(provider_id=self.PROVIDER_ID)
+
+    def test_credit_request_current_status(self):
+        request = CreditRequest.objects.create(
+            uuid="abcd1234",
+            username=self.USERNAME,
+            course=self.credit_course,
+            provider=self.credit_provider,
+        )
+
+        # Initially, status should be pending
+        self.assertEqual(request.current_status(), "pending")
+
+        # Update the status
+        CreditRequestStatus.objects.create(request=request, status="approved")
```

Comment:
A helper method on `CreditRequest` (e.g. `CreditRequest.set_status`) might prove useful for testing and in the future integration.